internal/domain/domain_operations.go

```diff
@@ -0,0 +1,242 @@
+/*
+ *  Copyright 2023 VMware, Inc.
+ *    SPDX-License-Identifier: MPL-2.0
+ */
+
+package domain
+
+import (
+	"context"
+	"fmt"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/vmware/terraform-provider-vcf/internal/cluster"
+	"github.com/vmware/terraform-provider-vcf/internal/constants"
+	"github.com/vmware/terraform-provider-vcf/internal/network"
+	validationUtils "github.com/vmware/terraform-provider-vcf/internal/validation"
+	"github.com/vmware/terraform-provider-vcf/internal/vcenter"
+	"github.com/vmware/vcf-sdk-go/client"
+	"github.com/vmware/vcf-sdk-go/client/clusters"
+	"github.com/vmware/vcf-sdk-go/client/domains"
+	"github.com/vmware/vcf-sdk-go/models"
+	"sort"
+)
+
+func CreateDomainCreationSpec(data *schema.ResourceData) (*models.DomainCreationSpec, error) {
```

Comment:
Do not see ssoDomainSpec and networkSeparationSpec?

Reply:
ssoDomainSpec is new in VCF 5.0, and I think Dimitar mentioned in another review that the vcf-go-sdk is still using the 4.5 Swagger spec.

networkSeparationSpec is not yet released if I recall correctly.